### DISCUSSION TITLE
Ping and Pong

### DISCUSSION THREAD

**Discussion by MiningInMySleep** [ctubio/Krypto-trading-bot#1069] [Q&A] [answered]
Hello,

Could we please have an exact definition of Ping and Pong because I clearly don't understand how it works.
This whole time I thought (assuming boomerang safety) that a ping is any filled order (buy or sell) that does not have a matching ping at least pongWidth away in profit. However, this is clearly not the case.

I've tried looking through trading-bot.data.h, but can't find anything relating to how boomerang operates.

Thanks!

**Comment by ctubio** [comment, marked as answer]
:koala: your description seems the case to me; the related code is all functions named `matchPong`

**Reply by MiningInMySleep** [reply]
The reason I'm asking is because I am seeing multiple pongs in K web UI but K log is showing them all as pings. For example on BTC/USD on Bitfinex I have a massive ping at 49k, pongWidth set to .1% and a K log full of pings at 50.7k which are limited by the trades/sec setting when they shouldn't be.

Edit: Images to illustrate the issue here.
![chrome_2021-02-24_21-26-36](https://user-images.githubusercontent.com/34825846/109000708-693c1400-76e7-11eb-9133-a875b60f5caa.png)
![chrome_2021-02-24_21-27-29](https://user-images.githubusercontent.com/34825846/109000714-6b9e6e00-76e7-11eb-9d8f-c5bec48ec3a5.png)

These pings should have been closed forever ago, but the majority (90+%) of the sells were pings in my K log (despite being pongs on web UI).

Edit 2: Stats are also very, very strange.
![chrome_2021-02-24_21-36-26](https://user-images.githubusercontent.com/34825846/109001585-74437400-76e8-11eb-90d0-abf0c94b86aa.png)

Edit 3: More background info. I have been running with ewmaPrice and stdev protections off. K UI latency is reporting ~30-40ms. Lifetime was set to 3000ms, so K was definitely taking trades, but not classifying them as pongs which I believe lead to them being limited by trades/sec.

Edit 4: It eventually started ponging, but well beyond the set pongWidth currently at .08%. Just to clarify, with boomerang, a pong should be makerFee% + pongWidth? Or are we doing 2*makerFee% + pongWidth? Either way, it seems to be ponging at much larger width when using %.

**Reply by ctubio** [reply]
yea it was a bug, (Quote object is correctly never destroyed but isPong status was wrongly never removed, so once a quote is a pong, all next quotes become also pongs)

after https://github.com/ctubio/Krypto-trading-bot/commit/4eca855131fc5dd3228528859e4e2bcd6fb5059b each niew quote will have the correct status (and limitations) of pings and pongs without matter what was the previous quote

**Comment by eniware-org** [comment]
I think it is related to the issue I reported in https://github.com/ctubio/Krypto-trading-bot/issues/1047#issuecomment-778682494
Also running a boomerang safety.

**Reply by ctubio** [reply]
:koala: many thanks @eniware-org for your help fixing this issue'¡

**Comment by Corfucinas** [comment]
Any updates on this?

**Reply by ctubio** [reply]
:koala: many thanks @Corfucinas for your help fixing this issue'¡
